rmw_connextdds_common/src/common/rmw_context.cpp

```diff
@@ -203,6 +203,26 @@ rmw_context_impl_t::initialize_node(
   // RMW_CONNEXT_ENV_DO_NOT_OVERRIDE_PUBLISH_MODE is empty.
   this->override_publish_mode = '\0' == do_not_override_publish_mode_env[0];
 
+  /* Lookup RMW_CONNEXT_ENV_ALLOW_TOPIC_QOS_PROFILES env variable.*/
+  const char * use_topic_profiles = nullptr;
+  lookup_rc = rcutils_get_env(
+    RMW_CONNEXT_ENV_ALLOW_TOPIC_QOS_PROFILES, &use_topic_profiles);
+
+  if (nullptr != lookup_rc || nullptr == qos_library) {
```

Comment:
The check should be for `use_topic_profiles` instead of `qos_library`

Reply:
Fixed in https://github.com/rticommunity/rmw_connextdds/pull/7/commits/09f7421f199f0aebb25655f5dc32931a0991b63a